ksql-engine/src/test/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNodeTest.java

```diff
@@ -255,14 +281,105 @@ public void shouldCreateSinkWithCorrectCleanupPolicyWindowedTable() {
 
     // Then:
     assertThat(stream, instanceOf(SchemaKTable.class));
-    verify(mockTopicClient).createTopic("output", 4, (short) 3, Collections.emptyMap());
+    verify(mockTopicClient).createTopic(SINK_KAFKA_TOPIC_NAME, 4, (short) 3, Collections.emptyMap());
   }
 
   @Test
   public void shouldCreateSinkWithCorrectCleanupPolicyStream() {
     // Then:
     assertThat(stream, instanceOf(SchemaKStream.class));
-    verify(mockTopicClient).createTopic("output", 4, (short) 3, Collections.emptyMap());
+    verify(mockTopicClient).createTopic(SINK_KAFKA_TOPIC_NAME, 4, (short) 3, Collections.emptyMap());
+  }
+
+  @Test
+  public void shouldCreateSinkWithTheSourcePartitionReplication() {
+    // Given:
+    createOutputNode(Collections.emptyMap(), true);
+
+    // When:
+    final SchemaKStream schemaKStream = buildStream();
+
+    // Then:
+    verify(mockTopicClient).createTopic(SINK_KAFKA_TOPIC_NAME, 1, (short) 2, Collections.emptyMap());
+    assertThat(schemaKStream, instanceOf(SchemaKStream.class));
+  }
+
+  @Test
+  public void shouldNotFetchSourceTopicPropsIfProvided() {
+    // Given:
+    createOutputNode(ImmutableMap.of(
+        KsqlConfig.SINK_NUMBER_OF_PARTITIONS_PROPERTY, 5,
+        KsqlConfig.SINK_NUMBER_OF_REPLICAS_PROPERTY, (short) 3
+    ), true);
+
+    // When:
+    final SchemaKStream schemaKStream = buildStream();
```

Comment:
nit:  following the pattern of the other tests:

```suggestion
   stream = buildStream();
```